hash/hash_test.go

```diff
@@ -0,0 +1,47 @@
+package hash
+
+import (
+	"crypto/subtle"
+	"encoding/base64"
+	"testing"
+
+	"gitlab.com/NebulousLabs/fastrand"
+)
+
+// TestDecodeHash is a short sanity check that ensures decodeHash works as
+// expected.
+func TestDecodeHash(t *testing.T) {
```

Comment:
This unit test is not covering any of the exit flows that aren't the happy flow.

Reply:
You definitely want to cover `ErrInvalidHash` and `ErrIncompatibleVersion` at least right?

Reply:
I added this quickly just to ensure the method does what I expect it to. And then I forgot to expand it. :)
More cases added.